pyfar/classes/transmission_matrix.py

```diff
@@ -825,6 +854,210 @@ def create_transmission_line(
+        propagation_direction: str = {'forwards', 'backwards'}
+            Defines the direction of sound propagation through the horn,
+            where ``'forwards'`` means from narrow to wide end
+            and ``'backwards'`` means from wide to narrow end.
+            Default is ``'forwards'``.
+
+        Returns
+        -------
+        pf.TransmissionMatrix
+            Transmission matrix for the conical horn section.
+
+        Example
+        -------
+        Arbitrarily sized lossless conical horn section.
+
+        .. plot::
+
+            >>> import pyfar as pf
+            >>> import numpy as np
+            >>> # Horn parameters
+            >>> S0 = 0.003
+            >>> S1 = 0.005
+            >>> L = 0.3
+            >>> frequencies = np.linspace(20, 20e3, 1000)
+            >>> omega = 2 * np.pi * frequencies
+            >>> k = pf.FrequencyData(
+            >>>    omega / pf.constants.reference_speed_of_sound, frequencies)
+            >>> Z0 = pf.constants.reference_air_impedance
+            >>> # Create the transmission matrix
+            >>> T = pf.TransmissionMatrix.create_conical_horn(
+            >>>    S0, S1, L, k, Z0, 'backwards')
+            >>> # Plot the transmission matrix
+            >>> pf.plot.freq(T.input_impedance(np.inf))
+
+        """
+        if not (isinstance(k, FrequencyData) or isinstance(k, Number)):
+            raise TypeError(
+                "The wave number k must be a float, complex,"
+                "or FrequencyData object.",
+            )
+        elif isinstance(k, FrequencyData):
+            frequencies = k.frequencies
+            k = k.freq
+        else:
+            if isinstance(k, complex):
+                k_re = k.real
+            else:
+                k_re = k
+            frequencies = (k_re * reference_speed_of_sound) / (2 * np.pi)
+        if isinstance(medium_impedance, FrequencyData):
+            if not np.allclose(
+                medium_impedance.frequencies,
+                frequencies,
+                atol=1e-15,
+            ):
+                raise ValueError(
+                    "The frequencies of characteristic_impedance must"
+                    "match those of k.",
+                )
+            else:
+                medium_impedance = medium_impedance.freq
+        elif not isinstance(medium_impedance, Number):
+            raise TypeError(
+                "The input medium_impedance must be a number"
+                "or a FrequencyData object.",
+            )
+        if not isinstance(propagation_direction, str):
+            raise TypeError(
+                "The input propagation_direction must be a string"
+                "with value 'forwards' or 'backwards'.",
+            )
+        if propagation_direction not in ("forwards", "backwards"):
+            raise ValueError(
+                "The string propagation_direction must either be"
+                "'forwards' or 'backwards'.",
```

Comment:
```suggestion
                "The string propagation_direction must either be "
                "'forwards' or 'backwards'.",
```
And another missing blank between two separate words